### PR TITLE
Use a mirror for registry.k8s.io to avoid hitting 5k request per minute limit in K8s scalability tests

### LINKED ISSUE #123672: [Failing Test] gce-master-scale-performance
### Which jobs are failing?

https://testgrid.k8s.io/sig-release-master-informing#gce-master-scale-performance

### Which tests are failing?

ci-kubernetes-e2e-gce-scale-performance.Overall
kubetest.ClusterLoaderV2

### Since when has it been failing?

02-25-24 to 02-27-24

latest since 03-03-24

### Testgrid link

https://testgrid.k8s.io/sig-release-master-informing#gce-master-scale-performance

### Reason for failure (if possible)

```
I0303 17:17:36.980573   17996 prometheus.go:439] Waiting for Prometheus stack to become healthy...
W0303 17:17:37.019005   17996 util.go:72] error while calling prometheus api: the server is currently unable to handle the request (get services http:prometheus-k8s:9090), response: k8s�

v1Status]

���Failure3no endpoints available for service "prometheus-k8s""ServiceUnavailable0��"�

      "involvedObject": {
        "kind": "StatefulSet",
        "namespace": "monitoring",
        "name": "prometheus-k8s",
        "uid": "9d65236a-7d47-4c6d-882d-807df0b57eed",
        "apiVersion": "apps/v1",
        "resourceVersion": "520714"
      },
      "reason": "FailedCreate",
      "message": "create Pod prometheus-k8s-0 in StatefulSet prometheus-k8s failed error: pods \"prometheus-k8s-0\" is forbidden: error looking up service account monitoring/prometheus-k8s: serviceaccount \"prometheus-k8s\" not found",
      "source": {
        "component": "statefulset-controller"
      },
      "firstTimestamp": "2024-03-03T17:17:32Z",
      "lastTimestamp": "2024-03-03T17:17:32Z",
      "count": 5,
      "type": "Warning",
      "eventTime": null,
      "reportingComponent": "statefulset-controller",
      "reportingInstance": ""
    }
    
    ```

### Anything else we need to know?

_No response_

### Relevant SIG(s)

/sig scalability

### PR DESCRIPTION
**Pull request by serathius**
#### What type of PR is this?
/kind flake

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubernetes/issues/123672

#### Special notes for your reviewer:

```release-note
NONE
```

/assign @BenTheElder

### CODE REVIEW

**Comment by serathius** [comment]
/sig scalability
/sig k8s-infra

/priority critical-urgent
Fixes flakes of scalability tests

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -3280,6 +3280,8 @@ oom_score = -999
 # See: https://github.com/kubernetes/k8s.io/issues/3411
 [plugins."io.containerd.grpc.v1.cri".registry.mirrors."k8s.gcr.io"]
   endpoint = ["https://registry.k8s.io", "https://k8s.gcr.io",]
+[plugins."io.containerd.grpc.v1.cri".registry.mirrors."registry.k8s.io"]
+  endpoint = ["https://mirror.gcr.io"]
```
Comment (wojtek-t) [inline cluster/gce/gci/configure-helper.sh:3284]: Just for my curiosity - how does it work with the line above? We're first mirroring k8s.gcr.io to registry.k8s.io and then mirroring registry.k8s.io to mirror.gcr.io ?

**Comment by ameukam** [comment]
I don't think a redirect to `mirror.gcr.io` will help in this case ? some of images needed. to run tests are only present in `registry.k8s.io` ?

**Comment by ameukam** [comment]
Or do we assume `mirror.gcr.io` will automatically cache any image at the first pull so we can avoid rate limiting ?

(Looking at https://explore.ggcr.dev/?repo=mirror.gcr.io/library/, I don't see `coredns` present which is needed to run the tests)

File: cluster/gce/gci/configure-helper.sh (hunk 1 above)
Comment (dims) [reply cluster/gce/gci/configure-helper.sh:3284]: we don't really need the 3281/3282 at this point right? (if there are still lingering usages of k8s.gcr.io, we can fix them when it fails)

**Comment by serathius** [comment]
> Or do we assume `mirror.gcr.io` will automatically cache any image at the first pull so we can avoid rate limiting ?

It should automatically cache https://cloud.google.com/container-registry/docs/pulling-cached-images

File: cluster/gce/gci/configure-helper.sh (hunk 1 above)
Comment (BenTheElder) [reply cluster/gce/gci/configure-helper.sh:3284]: > we don't really need the 3281/3282 at this point right? (if there are still lingering usages of k8s.gcr.io, we can fix them when it fails)

Yes, we can drop this, nothing should be pulling from k8s.gcr.io anymore.

Doesn't mirror.gcr.io only cache dockerhub?

**Comment by ameukam** [comment]
> > Or do we assume `mirror.gcr.io` will automatically cache any image at the first pull so we can avoid rate limiting ?
> 
> It should automatically cache https://cloud.google.com/container-registry/docs/pulling-cached-images

Only from Docker Hub ?
> Google Cloud services such as Cloud Build and Google Kubernetes Engine automatically check for cached images before attempting to pull an image from Docker Hub.

**Comment by serathius** [comment]
Closing for https://github.com/kubernetes/test-infra/pull/32476

**Pull request closed without merging**